components/api/core/api_impl.bal

```diff
@@ -499,30 +499,54 @@ returns http:Response {
     if (updateImageReq.hasHeader(constants:AUTHENTICATED_USER)) {
         string userId = updateImageReq.getHeader(constants:AUTHENTICATED_USER);
         log:printDebug(io:sprintf("%s is attempting to update image \'%s\' in organization %s", userId, imageName, orgName));
-        log:printDebug(updateImageBody.description);
-        sql:UpdateResult | error? updateImageRes = db:updateImage(orgName, imageName, updateImageBody.description, userId);
-
-        if (updateImageRes is sql:UpdateResult){
-            if (updateImageRes.updatedRowCount == 1) {
-                log:printDebug(io:sprintf("Image \'%s\' in organization \'%s\' is successfully updated. Author : %s", imageName, orgName, userId));
-                return buildSuccessResponse();
-            } else if (updateImageRes.updatedRowCount == 0){
-                log:printError(io:sprintf("Failed to update image \'%s\' in organization \'%s\' for Author %s : No matching records found",
-                imageName, orgName, userId));
+        log:printDebug(io:sprintf("Description : %s, Summary : %s", updateImageBody.description, updateImageBody.summary));
+
+        http:Response resp;
+        transaction {
+            var transactionId = transactions:getCurrentTransactionId();
+            log:printDebug("Started transaction " + transactionId + " for updating image " + imageName);
+
+            sql:UpdateResult | error? updateImageRes = db:updateImageDescriptionNSummary(orgName, imageName, updateImageBody.description, updateImageBody.summary, userId);
+            if (updateImageRes is sql:UpdateResult) {
+                if (updateImageRes.updatedRowCount == 1) {
+                    log:printDebug(io:sprintf("Description and summery of the image %s/%s is successfully updated. Author : %s", orgName, imageName, userId));
+                    error? updateImageKeywordsRes = db:updateImageKeywords(orgName, imageName, updateImageBody.keywords, userId);
+                    if updateImageKeywordsRes is error {
+                        log:printError(io:sprintf("Failed to update image %s/%s for Author %s.", orgName, imageName, userId), err = updateImageKeywordsRes);
+                        resp = buildErrorResponse(http:EXPECTATION_FAILED_417, constants:API_ERROR_CODE, "Unable to update image", "");
+                    } else {
+                        log:printDebug(io:sprintf("Successfully updated the keywords of the image %s/%s by %s", orgName, imageName, userId));
+                        resp = buildSuccessResponse();
+                    }
+                } else if (updateImageRes.updatedRowCount == 0) {
+                    log:printError(io:sprintf("Failed to update image %s/%s for Author %s : No matching records found",
+                    imageName, orgName, userId));
+                    resp = buildErrorResponse(http:EXPECTATION_FAILED_417, constants:API_ERROR_CODE, "Unable to update image", "");
```

Comment:
You should call "abort;" here

Reply:
IMO the error code should be 404.

Reply:
Since the error occurred because of not find any matching records, created a new API error code (ENTRY_NOT_FOUND_ERROR_CODE). Hence using buildErrorResponse()